`uk` locale: Function cityName not found in io.github.serpro69.kfaker.provider.Address

```
Caused by: java.util.NoSuchElementException: Function cityName not found in io.github.serpro69.kfaker.provider.Address@38a60bcb
        at io.github.serpro69.kfaker.FakerService.getFunctionName(FakerService.kt:587)
        at io.github.serpro69.kfaker.FakerService.access$getFunctionName(FakerService.kt:34)
        at io.github.serpro69.kfaker.FakerService$resolveExpression$resolvedExpression$1.invoke(FakerService.kt:497)
        at io.github.serpro69.kfaker.FakerService$resolveExpression$resolvedExpression$1.invoke(FakerService.kt:492)
        at io.github.serpro69.kfaker.FakerService.findMatchesAndAppendTail(FakerService.kt:622)
        at io.github.serpro69.kfaker.FakerService.resolveExpression-goSy63c(FakerService.kt:492)
        at io.github.serpro69.kfaker.FakerService.resolve(FakerService.kt:414)
        at io.github.serpro69.kfaker.provider.YamlFakeDataProvider$returnOrResolveUnique$3.invoke(YamlFakeDataProvider.kt:227)
        at io.github.serpro69.kfaker.provider.YamlFakeDataProvider$returnOrResolveUnique$3.invoke(YamlFakeDataProvider.kt:221)
        at io.github.serpro69.kfaker.provider.AbstractFakeDataProvider.resolveUniqueValue(AbstractFakeDataProvider.kt:74)
        at io.github.serpro69.kfaker.provider.AbstractFakeDataProvider.resolveUniqueValue(AbstractFakeDataProvider.kt:59)
        at io.github.serpro69.kfaker.provider.YamlFakeDataProvider.returnOrResolveUnique(YamlFakeDataProvider.kt:221)
        at io.github.serpro69.kfaker.provider.YamlFakeDataProvider.returnOrResolveUnique$default(YamlFakeDataProvider.kt:216)
        at io.github.serpro69.kfaker.provider.YamlFakeDataProvider.resolve(YamlFakeDataProvider.kt:77)
        at io.github.serpro69.kfaker.commerce.provider.Company.name(Company.kt:24)

```

Should be an easy fix, though I don't like modifying dict files really.... Need to find a better way to handle these discrepancies...

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -18,7 +18,9 @@
 [discrete]
 === Fixed
 
-* https://github.com/serpro69/kotlin-faker/issues/207[#207] (:core) Regexify generates invalid value 
+* https://github.com/serpro69/kotlin-faker/pull/240[#240] (:core) Fix NoSuchElementException in `uk.company.name`
+** Also fixes `uk.name.first_name` and `uk.name.last_name` to return both male and female names.
+* https://github.com/serpro69/kotlin-faker/issues/207[#207] (:core) Regexify generates invalid value
 * https://github.com/serpro69/kotlin-faker/issues/208[#208] (:core) Regexify fails with StackOverflowError
 
 [discrete]

--- a/core/src/integration/kotlin/io/github/serpro69/kfaker/provider/AddressIT.kt
+++ b/core/src/integration/kotlin/io/github/serpro69/kfaker/provider/AddressIT.kt
@@ -21,6 +21,14 @@ class AddressIT : DescribeSpec({
                     address("uk").citySuffix() shouldBe ""
                 }
             }
+
+            context("non-default key reference - gh #239") {
+                repeat(10) {
+                    it("city() does NOT throw NoSuchElementException run#$it") {
+                        shouldNotThrow<NoSuchElementException> { address("uk").city() }
+                    }
+                }
+            }
         }
 
         context("nb-NO locale") {

--- a/core/src/integration/kotlin/io/github/serpro69/kfaker/provider/NameIT.kt
+++ b/core/src/integration/kotlin/io/github/serpro69/kfaker/provider/NameIT.kt
@@ -9,6 +9,13 @@ class NameIT : DescribeSpec({
     describe("Name provider") {
         val name: (locale: String) -> Name = { faker { fakerConfig { locale = it } }.name }
 
+        context("uk locale") {
+            it("generates a name") {
+                val names = List(42) { name("uk").name() }
+                names shouldNotContain ""
+            }
+        }
+
         context("ru locale") {
             it("generates lastName") {
                 val lastNames = List(42) { name("ru").lastName() }

--- a/core/src/main/resources/locales/uk.yml
+++ b/core/src/main/resources/locales/uk.yml
@@ -281,9 +281,7 @@ uk:
         - Стрийська
         - Вузька
         - Староміська
-      # NB! don't remove city_name anchor
-      # See https://github.com/serpro69/kotlin-faker/issues/239
-      city_name: &city_name
+      city_name:
         - Алчевськ
         - Артемівськ
         - Бердичів
@@ -343,9 +341,8 @@ uk:
         - Ялта
       city_prefix: []
       city_suffix: []
-      # NB! don't change anchor reference
-      # See https://github.com/serpro69/kotlin-faker/issues/239
-      city: *city_name
+      city:
+        - "#{city_name}"
       street_name:
         - "#{feminine_street_prefix} #{feminine_street_title}"
         - "#{masculine_street_prefix} #{masculine_street_title}"

--- a/core/src/main/resources/locales/uk.yml
+++ b/core/src/main/resources/locales/uk.yml
@@ -281,7 +281,9 @@ uk:
         - Стрийська
         - Вузька
         - Староміська
-      city_name:
+      # NB! don't remove city_name anchor
+      # See https://github.com/serpro69/kotlin-faker/issues/239
+      city_name: &city_name
         - Алчевськ
         - Артемівськ
         - Бердичів
@@ -341,8 +343,9 @@ uk:
         - Ялта
       city_prefix: []
       city_suffix: []
-      city:
-        - "#{city_name}"
+      # NB! don't change anchor reference
+      # See https://github.com/serpro69/kotlin-faker/issues/239
+      city: *city_name
       street_name:
         - "#{feminine_street_prefix} #{feminine_street_title}"
         - "#{masculine_street_prefix} #{masculine_street_title}"
@@ -1784,8 +1787,8 @@ uk:
         - "#{prefix} #{Name.female_first_name}"
         - "#{prefix} #{Name.male_first_name}"
         - "#{prefix} #{Name.male_last_name}"
-        - "#{prefix} #{Address.city_name}#{product}#{suffix}"
-        - "#{prefix} #{Address.city_name}#{suffix}"
+        - "#{prefix} #{Address.city}#{product}#{suffix}" # NB! do not change, see #239
+        - "#{prefix} #{Address.city}#{suffix}" # NB! do not change, see #239
     music:
       instruments:
         - Електрична гітара

--- a/core/src/main/resources/locales/uk.yml
+++ b/core/src/main/resources/locales/uk.yml
@@ -420,7 +420,7 @@ uk:
         - zt.ua
         - укр
     name:
-      male_first_name: &1
+      male_first_name: &male_first_name # NB! do not change anchor name
         - Августин
         - Аврелій
         - Адам
@@ -733,7 +733,7 @@ uk:
         - Юрійович
         - Юхимович
         - Ярославович
-      male_last_name: &2
+      male_last_name: &male_last_name # NB! do not change anchor name
         - Андрухович
         - Бабух
         - Балабан
@@ -973,7 +973,7 @@ uk:
         - Яцишин
         - Яцьків
         - Ящук
-      female_first_name:
+      female_first_name: &female_first_name # NB! do not change anchor name
         - Аврелія
         - Аврора
         - Агапія
@@ -1284,7 +1284,7 @@ uk:
         - Юріївна
         - Юхимівна
         - Ярославівна
-      female_last_name:
+      female_last_name: &female_last_name # NB! do not change anchor name
         - Андрухович
         - Бабух
         - Балабан
@@ -1515,8 +1515,12 @@ uk:
         - Яцишина
         - Яцьків
         - Ящук
-      first_name: *1
-      last_name: *2
+      first_name: # NB! do not change anchor references, we should include both female and male first names here
+        - *male_first_name
+        - *female_first_name
+      last_name: # NB! do not change anchor references, we should include both female and male last names here
+        - *male_last_name
+        - *female_last_name
       name:
         - "#{male_first_name} #{male_last_name}"
         - "#{male_last_name} #{male_first_name}"
@@ -1784,9 +1788,8 @@ uk:
         - скло
         - дерево
       name:
-        - "#{prefix} #{Name.female_first_name}"
-        - "#{prefix} #{Name.male_first_name}"
-        - "#{prefix} #{Name.male_last_name}"
+        - "#{prefix} #{Name.first_name}" # NB! do not change, see #239
+        - "#{prefix} #{Name.last_name}" # NB! do not change, see #239
         - "#{prefix} #{Address.city}#{product}#{suffix}" # NB! do not change, see #239
         - "#{prefix} #{Address.city}#{suffix}" # NB! do not change, see #239
     music:

--- a/faker/commerce/src/integration/kotlin/io/github/serpro69/kfaker/commerce/provider/CompanyIT.kt
+++ b/faker/commerce/src/integration/kotlin/io/github/serpro69/kfaker/commerce/provider/CompanyIT.kt
@@ -24,13 +24,3 @@ class CompanyIT : DescribeSpec({
         }
     }
 })
-
-fun main() {
-    val f = io.github.serpro69.kfaker.faker { fakerConfig { locale = "uk" } }
-    repeat(100) {
-        println("""
-        ${f.name.firstName()} ${f.name.lastName()}
-        """
-        )
-    }
-}

--- a/faker/commerce/src/integration/kotlin/io/github/serpro69/kfaker/commerce/provider/CompanyIT.kt
+++ b/faker/commerce/src/integration/kotlin/io/github/serpro69/kfaker/commerce/provider/CompanyIT.kt
@@ -9,6 +9,12 @@ class CompanyIT : DescribeSpec({
     describe("Company Provider") {
         val company: (locale: String) -> Company = { faker { fakerConfig { locale = it } }.company }
 
+        context("uk locale") {
+            it("should generate a valid name") {
+                shouldNotThrow<NoSuchElementException> { company("uk").name() shouldNotBe "" }
+            }
+        }
+
         context("ja locale") {
             it("should generate a valid name") {
                 shouldNotThrow<NoSuchElementException> { company("ja").name() shouldNotBe "" }

--- a/faker/commerce/src/integration/kotlin/io/github/serpro69/kfaker/commerce/provider/CompanyIT.kt
+++ b/faker/commerce/src/integration/kotlin/io/github/serpro69/kfaker/commerce/provider/CompanyIT.kt
@@ -10,8 +10,10 @@ class CompanyIT : DescribeSpec({
         val company: (locale: String) -> Company = { faker { fakerConfig { locale = it } }.company }
 
         context("uk locale") {
-            it("should generate a valid name") {
-                shouldNotThrow<NoSuchElementException> { company("uk").name() shouldNotBe "" }
+            repeat(10) {
+                it("should generate a valid name run#$it") {
+                    shouldNotThrow<NoSuchElementException> { company("uk").name() shouldNotBe "" }
+                }
             }
         }
 
@@ -22,3 +24,13 @@ class CompanyIT : DescribeSpec({
         }
     }
 })
+
+fun main() {
+    val f = io.github.serpro69.kfaker.faker { fakerConfig { locale = "uk" } }
+    repeat(100) {
+        println("""
+        ${f.name.firstName()} ${f.name.lastName()}
+        """
+        )
+    }
+}

